libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -263,6 +264,15 @@ class MapboxNavigation(
         }
     }
 
+    /**
+     * @see [startTripSession]
+     *
+     * @param locationPollingOptions override the frequencies for updating navigator data.
+     */
+    fun startTripSession(locationPollingOptions: LocationPollingOptions) {
```

Comment:
Why not exposing the `LocationPollingOptions` through the `NavigationOptions`? That way this wouldn't be needed.

Reply:
we can't update `NavigationOptions` between trips. have to restart the app to change it

Reply:
Is that an option we want to expose though? Currently all the options are configured like that for the whole `MapboxNavigation` lifecycle 🤔

Reply:
For example, we'll never be able to do any battery optimization if the polling rates are configured during construction
https://github.com/mapbox/navigation-sdks/issues/100

Reply:
Is that a topic for a broader discussion? Maybe something to consider for `2.0` as it implies changes in the APIs and how they're / should be designed?

Reply:
To be fair, I think whatever we do here will be a part of a broader discussion to consider for `2.0`.

So this change is looking for stability, and being able to backport it to `1.2`